krm-functions/lib/kptrl/kptrl_test.go

```diff
@@ -0,0 +1,342 @@
+			}
+			got := r.GetObjects(tObj)
+
+			if got == nil {
+				if tc.want != nil {
+					t.Errorf("TestGetObject: -want: %v, +got:%v\n", tc.want, got)
+				}
+			} else {
+				objRef := &corev1.ObjectReference{
+					APIVersion: got[0].GetAPIVersion(),
+					Kind:       got[0].GetKind(),
+					Name:       got[0].GetName(),
+				}
+				if diff := cmp.Diff(tc.want, objRef); diff != "" {
+					t.Errorf("TestParseObjectKind: -want, +got:\n%s", diff)
+				}
+			}
+
+		})
+	}
+}
+
+func TestGetObjects(t *testing.T) {
+	rl, err := fn.ParseResourceList(resList)
+	if err != nil {
+		t.Errorf("cannot parse resourceList: %s", err.Error())
+	}
+	r := &ResourceList{
+		*rl,
+	}
+
+	cases := map[string]struct {
+		wantLen         int
+		wantAPIVersions []string
+	}{
+		"GetObjects": {
+			wantLen: 2,
+			wantAPIVersions: []string{
+				"a.a/v1",
+				"b.b/v1",
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			objs := r.GetAllObjects()
+
+			if len(objs) != tc.wantLen {
+				t.Errorf("TestGetObjects: -want %d, +got: %d\n", tc.wantLen, len(objs))
+			}
+
+			for i, o := range objs {
+				if o.GetAPIVersion() != tc.wantAPIVersions[i] {
+					t.Errorf("TestGetObjects: -want %s, +got: %s\n", o.GetAPIVersion(), tc.wantAPIVersions[i])
+				}
+			}
+		})
+	}
+}
+
+func TestSetObject(t *testing.T) {
+	rl, err := fn.ParseResourceList(resList)
+	if err != nil {
+		t.Errorf("cannot parse resourceList: %s", err.Error())
+	}
+	r := &ResourceList{
+		*rl,
+	}
+
+	cases := map[string]struct {
+		t    []byte
+		want *corev1.ObjectReference
+	}{
+		"SetObjectNew": {
+			t: objC,
+		},
+		"SetObjectOverwrite": {
+			t: objA,
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			tObj, err := fn.ParseKubeObject(tc.t)
+			if err != nil {
+				t.Errorf("cannot parse test obj: %s", err.Error())
+			}
+
+			r.SetObject(tObj)
+			got := r.GetObjects(tObj)
+
+			if got == nil {
+				t.Errorf("TestGetObject: -want: %v, +got:%v\n", tObj, got)
+			} else {
+				if diff := cmp.Diff(tObj.GetLabels(), got[0].GetLabels()); diff != "" {
+					t.Errorf("TestParseObjectKind: -want, +got:\n%s", diff)
+				}
+			}
+		})
+	}
+}
+
+func TestDeleteObject(t *testing.T) {
+	rl, err := fn.ParseResourceList(resList)
+	if err != nil {
+		t.Errorf("cannot parse resourceList: %s", err.Error())
+	}
+	r := &ResourceList{
+		*rl,
+	}
+
+	cases := map[string]struct {
+		t    []byte
+		want *corev1.ObjectReference
+	}{
+		"DeleteObj": {
+			t: objB,
+		},
+		"DeleteNonExistingObject": {
+			t: objC,
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			tObj, err := fn.ParseKubeObject(tc.t)
+			if err != nil {
+				t.Errorf("cannot parse test obj: %s", err.Error())
+			}
+
+			r.DeleteObject(tObj)
+			got := r.GetObjects(tObj)
+
+			if got != nil {
+				t.Errorf("TestDeleteObject: -want: nil, +got:%v\n", got)
+			}
+		})
+	}
+}
+
+func TestConurrency(t *testing.T) {
```

Comment:
This is not needed now as we have removed the mutex.

Reply:
agreed and removed